### ISSUE TITLE
when NUM_OF_CLASSES=2,loss doesn't work?

### ISSUE THREAD

**Issue by lily10086** [DeepSegment/ConvNet-Segment#10]
loss = tf.reduce_mean((tf.nn.sparse_softmax_cross_entropy_with_logits(logits=logits,
                                                                          labels=tf.squeeze(annotation, squeeze_dims=[3]),
                                                                          name="entropy")))
when classes is 2,I try train it ,But the loss is not normal,How can adapt it??

**Comment by txzhao** [comment]
Hi,

What do you mean by not normal? Can you elaborate it or could you show me the loss graph?

**Comment by lily10086** [comment]
I think  tf.nn.sparse_softmax_cross_entropy_with_logits can't be used  when num_classes is 2.
loss = tf.reduce_mean((tf.nn.sparse_softmax_cross_entropy_with_logits(logits=logits,
                                                                          labels=tf.squeeze(annotation, squeeze_dims=[3]),
                                                                          name="entropy")))

**Comment by txzhao** [comment]
According to the doc of tensorflow, _tf.nn.sparse_softmax_cross_entropy_with_logits_ should work fine when dealt with the 2-class case. Can you double check the size of your arguments? And what's the error information?

https://www.tensorflow.org/versions/r0.12/api_docs/python/nn/classification

**Comment by liuzhisheng1226** [comment]
我用多类训练的时候没有问题,但训练只有两类的时候出了问题,所以我怀疑这个loss不适合两类的...

**Comment by txzhao** [comment]
出问题是指报错吗？报错信息是什么？

**Comment by liuzhisheng1226** [comment]
没报错，就是loss输出感觉不正常。
现在只有loss输出，能不能增加些别的指标啊，比如正确率等




发自网易邮箱大师
在2018年01月04日 09:56，Tianxiao Zhao 写道：

出问题是指报错吗？报错信息是什么？

**Comment by txzhao** [comment]
其他的指标比如原论文中的那几个metrics是有计算的 要切换到test模式下运行 结果存到了metrics.txt文件中 (inception_FCN.py, line227-272)

**Comment by lily10086** [comment]
嗯嗯,我又训练了一次,没有问题,效果还不错,是我以前训练方法有问题.
谢谢你!!!

2018-01-04 10:03 GMT+08:00 Tianxiao Zhao <notifications@github.com>:

**Comment by txzhao** [comment]
好的，不用谢！
